File: backend/schemas/constraint.py
from typing import Optional, List
from pydantic import Field
from backend.schemas.base import BaseSchema


class ConstraintCreateRequest(BaseSchema):
    type: str = Field(..., alias="type")
    source_id: Optional[int] = Field(None, alias="sourceId")
    target_id: Optional[int] = Field(None, alias="targetId")
    value: Optional[float] = Field(None, alias="value")
    operator: Optional[str] = Field(None, alias="operator")
    priority: Optional[str] = Field(None, alias="priority")


class ConstraintResponse(BaseSchema):
    id: int
    type: str = Field(..., alias="type")
    source_id: Optional[int] = Field(None, alias="sourceId")
    target_id: Optional[int] = Field(None, alias="targetId")
    value: Optional[float] = Field(None, alias="value")
    operator: Optional[str] = Field(None, alias="operator")
    priority: Optional[str] = Field(None, alias="priority")


class ConstraintListResponse(BaseSchema):
    constraints: List[ConstraintResponse]

File: frontend/src/types/constraint.ts
export interface Constraint {
  id: number;
  type: string;
  sourceId?: number | null;
  targetId?: number | null;
  value?: number | null;
  operator?: string | null;
  priority?: string | null; // "hard" | "soft"
}

export interface ConstraintCreateRequest {
  type: string;
  sourceId?: number | null;
  targetId?: number | null;
  value?: number | null;
  operator?: string | null;
  priority?: string | null;
}

export interface ConstraintResponse extends Constraint {}

export interface ConstraintListResponse {
  constraints: Constraint[];
}
